app/controllers/hq/partners_controller.rb

```diff
@@ -1,71 +1,44 @@
 class Hq::PartnersController < ApplicationController
-  before_action :authenticate_admin_user!
+  before_action :authenticate_admin_user!,
+                :new_partner_facade
   layout 'application'
 
   def index
-    @partners= Partner.all
+    @partners = Partner.all
   end
 
   def new
-    @partner = Partner.new
-    load_associations
+    @partner_facade.build_partner
   end
   
   def create 
-    build_partner
-    save_partner or re_render 'new'
+    @partner_facade.build_partner(params[:partner])
```

Comment:
Only problem is that for the a successful `build_partner!(args)` we also need to set the `flash[:successful]` message and we don't have access to that from the facade. Same with `redirect_to`.